frontend/src/api/Socket.ts

```diff
@@ -3,14 +3,52 @@ import Stomp from 'stompjs';
 
 let stompClient:any = null;
 
-export const connect = (endpoint:string) => {
-  const socket:any = new SockJS(endpoint);
-  stompClient = Stomp.over(socket);
-  stompClient.connect();
+type User = {
+  nickname: string;
+}
+
+// Variable to hold the current connected state.
+let connected = false;
+
+// Create constants for the subscription and send message URLs.
+const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
+const SEND_USER_URL:string = '/api/v1/socket/user';
+
+export const sendUser = (nickname:string) => {
+  if (connected) {
+    const nicknameInput:string = (nickname !== '') ? nickname : 'Anonymous';
+    stompClient.send(SEND_USER_URL, {}, nicknameInput);
+  } else {
+    console.error('You must be connected to a socket before sending user information.');
```

Comment:
Note for future: these might eventually have to be converted to some kind of UI error message

Reply:
Yeah that makes sense -- I added these `console.error(...)` messages because the UI that is currently setup in this PR will be removed (we won't have "Connect", "Disconnect", and "Send User Information" buttons). I'll be sure to update these to show the relevant information in the UI when or if necessary; I will note that after I update the UI, the user will not be able to connect to the socket when already connected or vice versa, so in that case I would leave this as-is as no UI error message would be needed. I'll address that again when the time comes, though.